src/credential/sd_jwt.rs

```diff
@@ -0,0 +1,364 @@
+use std::sync::Arc;
+
+use json_syntax::Object;
+use oid4vp::core::{
+    credential_format::ClaimFormatDesignation, presentation_definition::PresentationDefinition,
+};
+use ssi::{
+    claims::{
+        jwt::{AnyClaims, JWTClaims},
+        sd_jwt::{RevealedSdJwt, SdJwtBuf},
+        vc::v2::{Credential as _, JsonCredential},
+    },
+    prelude::AnyJsonCredential,
+};
+use uuid::Uuid;
+
+use crate::{CredentialType, KeyAlias};
+
+use super::{jwt_vc::JwtVc, Credential, ParsedCredential, ParsedCredentialInner};
+
+#[derive(Debug, uniffi::Object)]
+pub struct SdJwt {
+    pub(crate) id: Uuid,
+    pub(crate) key_alias: Option<KeyAlias>,
+    pub(crate) revealed_claims: JWTClaims,
+    pub(crate) credential: JsonCredential<Object>,
+    pub(crate) inner: SdJwtBuf,
+}
+
+// Internal utility methods for decoding a SdJwt.
+impl SdJwt {
+    /// Decode a SdJwt instance and return the revealed claims as a JSON value.
+    pub fn decode_reveal_json(&self) -> Result<serde_json::Value, SdJwtError> {
+        serde_json::to_value(&self.revealed_claims)
+            .map_err(|e| SdJwtError::Serialization(e.to_string()))
+    }
```

Comment:
No decoding or revealing is happening, this could be misleading.
```suggestion
    /// Return the revealed claims as a JSON value.
    pub fn revealed_claims_as_json(&self) -> Result<serde_json::Value, SdJwtError> {
        serde_json::to_value(&self.revealed_claims)
            .map_err(|e| SdJwtError::Serialization(e.to_string()))
    }
```

Reply:
I've updated these changes, but was unable to commit your suggested changes. 🙏